app/uk/gov/hmrc/timetopay/arrangement/services/PollerService.scala

```diff
@@ -34,55 +34,65 @@ import scala.util.{Failure, Success}
 
 @Singleton
 class PollerService @Inject() (
-    actorSystem:                       ActorSystem,
     desArrangementApiServiceConnector: DesArrangementApiServiceConnector,
     ttpArrangementRepositoryWorkItem:  TTPArrangementWorkItemRepository,
     queueConfig:                       QueueConfig,
     crypto:                            CryptoService,
     auditService:                      AuditService,
-    val clock:                         Clock)(
+    val clock:                         Clock,
+    scheduler:                         Scheduler,
+    onCompleteAction:                  OnCompleteAction
+)(
     implicit
     ec: ExecutionContext
-) extends ExclusiveScheduledJob {
+) {
 
   private val logger = QueueLogger(this.getClass)
   val initialDelay: FiniteDuration = queueConfig.initialDelay
   val interval: FiniteDuration = queueConfig.interval
 
-  override def executeInMutex(implicit ec: ExecutionContext): Future[Result] = {
-    ttpArrangementRepositoryWorkItem.findAll().map{ ls =>
-      val txt = ls.map(x => x.status).groupBy(x => x).map{ x => s"${x._1.toString}: ${x._2.size.toString}" }.mkString(", ")
-      logger.track("Retry poller - WorkItem counts- " + txt)
-    }
+  val name: String = "TTP Arrangement Poller"
 
-    process().map(_ => Result(""))
+  private def scheduleNext(interval: FiniteDuration): Unit = {
```

Comment:
scheduling is now done one at a time at the start and just after each scheduled run has finished  instead of at regular intervals - this will eliminate the scenario where two scheduled runs happen at the same time.